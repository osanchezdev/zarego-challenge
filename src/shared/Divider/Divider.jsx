import React from 'react';
import { string } from 'prop-types';
import styles from './Divider.module.css';

const Divider = ({ text }) => (
  <div className={styles.divider__wrapper}>
    {text && (
      <div className={styles.divider_text__wrapper}>
        <span className={styles.divider_text}>{text}</span>
      </div>
    )}
    <div role="separator" className={styles.divider} />
  </div>
);

Divider.propTypes = {
  text: string,
};

Divider.defaultProps = {
  text: null,
};

export default Divider;
