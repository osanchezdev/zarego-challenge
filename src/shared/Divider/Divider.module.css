.divider__wrapper {
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.divider_text {
  margin-right: 25px;
  font-size: 24px;
}

.divider {
  flex: 1 1 auto;
  height: 2px;
  background: #c1c1c1;
}
