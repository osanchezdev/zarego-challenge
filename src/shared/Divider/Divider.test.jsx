import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Divider from './Divider';

describe('Divider component snapshot', () => {
  it('should render component', () => {
    const tree = renderer.create(<Divider />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Divider component funtional tests', () => {
  it('should not render text, only separator line', () => {
    const dividerText = 'This is a test';
    render(<Divider />);

    const separatorLine = screen.getByRole('separator');
    const text = screen.queryByText(dividerText);
    expect(text).not.toBeInTheDocument();
    expect(separatorLine).toBeInTheDocument();
  });

  it('should render text', () => {
    const dividerText = 'This is a test';
    render(<Divider text={dividerText} />);

    const text = screen.queryByText(dividerText);
    expect(text).toBeInTheDocument();
  });
});
