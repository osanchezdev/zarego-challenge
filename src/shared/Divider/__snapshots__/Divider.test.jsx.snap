// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider component snapshot should render component 1`] = `
<div
  className="divider__wrapper"
>
  <div
    className="divider"
    role="separator"
  />
</div>
`;
